backend/services/grading_embedding_service.py

```
@@ -0,0 +1,388 @@
+                # Create overlap from last few sentences
+                overlap_sentences = current_chunk.split('.')[-3:]  # Last 3 sentences
+                current_chunk = '. '.join(overlap_sentences) + '. ' + sentence
+            else:
+                current_chunk += '. ' + sentence if current_chunk else sentence
+        
+        if current_chunk.strip():
+            chunks.append(current_chunk.strip())
+        
+        return chunks
+    
+    async def _get_embedding_async(self, text: str, cache_key: str) -> List[float]:
```

Ruff (0.14.1)

Unused method argument: `cache_key`

(ARG002)